### PR TITLE
feat: [PL-24488]: CG - Populate email for user when approval is done via API Key

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
Populate email for user object when approval is done via API Key, on the GraphQL approvals API.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38013)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38013)

**Comment by vcalasansh** [comment]
trigger codebasehash,messagecheck

File: 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java [hunk 1]
```diff
@@ -158,7 +158,11 @@ private ApprovalDetails constructApprovalDetailsFromInput(
     EmbeddedUser user = null;
     boolean isApprovedViaApiKey = false;
     if (apiKeyEntry != null) {
-      user = EmbeddedUser.builder().uuid(apiKeyEntry.getUuid()).name(apiKeyEntry.getName()).build();
+      user = EmbeddedUser.builder()
+                 .uuid(apiKeyEntry.getUuid())
+                 .name(apiKeyEntry.getName())
```
Comment (abhinav-harness) [inline 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java:163]: will name be righet here?

Comment (vcalasansh) [reply 380-cg-graphql/src/main/java/software/wings/graphql/datafetcher/approval/ApproveOrRejectApprovalsDataFetcher.java:163]: Thank you for the comment! Yes, the name will be the API key's name. If user who approved with the API key provides his e-mail, we will show it too. From what I understood in our discussions, this is the desired behavior. Please let me know what you think about this.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38013)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by lasdox — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38013)* status: :shipit: complete! ✓

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38013)* status: :shipit: complete! ✓

**Pull request merged**